### ISSUE TITLE
Can not open devcontainers through VSCode, only through attach to terminal

### ISSUE THREAD

**Issue by ChristopherCapito** [microsoft/vscode-remote-release#8284]
- VSCode Version: 1.76.2
- Local OS Version: Windows 11 Pro 10.0.22621 Build 2621
- Remote OS Version: Debian 10 (buster)
- Remote Extension/Connection Type: Containers/WSL/Server

- Logs:
[6326 ms] Dev Containers 0.287.0 in VS Code 1.76.2 (ee2b180d582a7f601fa6ecfdad8d9fd269ab1884).
[6326 ms] Start: Run: ssh 10.169.1.4 /bin/sh
[6336 ms] Start: Run in host: id -un

Steps to Reproduce:

1. Create a VM on Azure
2. Connect to the VM via Azure Bastion SSH Tunnelling
3. Try to reopen the container in a devcontainer

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

Greetings. I am trying to connect my local VSCode to a devcontainer running in an Azure VM. I am getting my SSH credentials for AAD based linux login via "az ssh config" and i am then connecting to the host via "az network bastion tunnel" because bastion and AAD login are a requirement.

I can also build, execute and run the devcontainer from the remote terminal. However, trying to attach VSCode does not work. Reopen in Container does not work. All I am getting is a forever loading "Setting up Dev Containers (show log)" and the following in said logs:

[6326 ms] Dev Containers 0.287.0 in VS Code 1.76.2 (ee2b180d582a7f601fa6ecfdad8d9fd269ab1884).
[6326 ms] Start: Run: ssh 10.169.1.4 /bin/sh
[6336 ms] Start: Run in host: id -un

Afterwards, nothing happens. For reference, this is my SSH Config on my local machine (name changed)

```
Host *
	ForwardAgent yes

Host Azure
	User username@domain
	HostName 10.169.1.4
	CertificateFile "path to my file"
	IdentityFile "path to my file"

Host 10.169.1.4
	User username@domain
	HostName 127.0.0.1
	CertificateFile "path to my file"
	IdentityFile "path to my file"
	StrictHostKeyChecking=No
    UserKnownHostsFile=\\.\NUL
	Port 22

```

The logs do not point to the isssue sadly. It seems like the process is just stuck somewhere without error. Has anyone experienced this so far?

**Comment by chrmarti** [comment]
Does `ssh 10.169.1.4 /bin/sh` work when you run it in a terminal? It looks like that might be stuck.

**Comment by ChristopherCapito** [comment]
I ran the command again with a verbose log. I have obfuscated the login and the key portion.

v 10.169.1.4 /bin/sh
OpenSSH_7.9p1 Debian-10+deb10u2, OpenSSL 1.1.1n  15 Mar 2022
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: Connecting to 10.169.1.4 [10.169.1.4] port 22.
debug1: Connection established.
debug1: identity file /home/myusername/.ssh/id_rsa type -1
debug1: identity file /home/myusername/.ssh/id_rsa-cert type -1
debug1: identity file /home/myusername/.ssh/id_dsa type -1
debug1: identity file /home/myusername/.ssh/id_dsa-cert type -1
debug1: identity file /home/myusername/.ssh/id_ecdsa type -1
debug1: identity file /home/myusername/.ssh/id_ecdsa-cert type -1
debug1: identity file /home/myusername/.ssh/id_ed25519 type -1
debug1: identity file /home/myusername/.ssh/id_ed25519-cert type -1
debug1: identity file /home/myusername/.ssh/id_xmss type -1
debug1: identity file /home/myusername/.ssh/id_xmss-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_7.9p1 Debian-10+deb10u2
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.9p1 Debian-10+deb10u2
debug1: match: OpenSSH_7.9p1 Debian-10+deb10u2 pat OpenSSH* compat 0x04000000
debug1: Authenticating to 10.169.1.4:22 as 'myusername'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ecdsa-sha2-nistp256
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:wpoj562poij5po6h5upo2h6upio2h536ipouh652op3jhoi
debug1: Host '10.169.1.4' is known and matches the ECDSA host key.
debug1: Found key in /home/myusername/.ssh/known_hosts:1
debug1: rekey after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey after 134217728 blocks
debug1: Will attempt key: /home/myusername/.ssh/id_rsa 
debug1: Will attempt key: /home/myusername/.ssh/id_dsa 
debug1: Will attempt key: /home/myusername/.ssh/id_ecdsa 
debug1: Will attempt key: /home/myusername/.ssh/id_ed25519 
debug1: Will attempt key: /home/myusername/.ssh/id_xmss 
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey
debug1: Next authentication method: publickey
debug1: Trying private key: /home/myusername/.ssh/id_rsa
debug1: Trying private key: /home/myusername/.ssh/id_dsa
debug1: Trying private key: /home/myusername/.ssh/id_ecdsa
debug1: Trying private key: /home/myusername/.ssh/id_ed25519
debug1: Trying private key: /home/myusername/.ssh/id_xmss
debug1: No more authentication methods to try.
myusername@10.169.1.4: Permission denied (publickey).

**Comment by ChristopherCapito** [comment]
I will try that tomorrow. Thank you.

Since the setup I am trying to use is a bit involved (AADSSHLogin tunnelled via Azure Bastion), I created a demo machine that is very basic. It is just a VM with Debian 11, VSCode and Docker.

I added a public IP, so no Bastion is needed. Here, all VSCode functionality works as expected, and the devcontainer build performance is also good at around max. 2 minutes for a simple container. I am required to use Bastion however, due to networking policies.
